config/samples/csharp_example_loadtest.yaml

```diff
@@ -24,10 +24,13 @@ spec:
     name: '0'
     run:
       args:
-      - exec
-      - qps_worker/Grpc.IntegrationTesting.QpsWorker.dll
+      - -c
+      - |
+        timeout --kill-after=$${KILL_AFTER} $${POD_TIMEOUT} dotnet exec
```

Comment:
improvement opportunity: I think just modifying the command to  run worker wrapped with timeout invocation works, but the downside is that all users need to remember to also invoke `timeout --kill-after=$${KILL_AFTER} $${POD_TIMEOUT}` in any custom experiments they run (e.g. experimenting with new images, languages, etc. where you need to specify the run worker command), otherwise they are risking their experiment will create pod don't terminate correctly.

So it would be nice if there was a way for the framework to automatically inject the `timeout --kill-after=$${KILL_AFTER} $${POD_TIMEOUT}` wrapper, without making that part of the worker run command explicitly.

Idea: docker images have the concept of `ENTRYPOINT` (which is simply a way to prepend any commands executed by the docker image with a command. https://docs.docker.com/engine/reference/builder/#entrypoint

so perhaps our Dockerimages can specify
```
ENTRYPOINT ["timeout", "--kill-after=${KILL_AFTER}", "${POD_TIMEOUT}"]
``` 

which would then inject the timeout logic automatically, without needing to make the run worker commands ugly
(and also needing to use the "bash -c" wrapping with confusing `$$`).
note that I haven't tried this, but it seems like something that would work.

Reply:
I think it could work. We could also try to add `"timeout", "--kill-after=${KILL_AFTER}", "${POD_TIMEOUT}"` as a prefix  for the command when we created the pods from controller. Both of them would work fine when the start-worker-commands are exposed in the loadtest configuration file, but like php7, where we have to wrap the start-worker-commands with a script, both of our ways would put the timeout on the wrapper.

Reply:
We are moving forward on the ENTRYPOINT method you suggested, let me verify and update the pr

Reply:
I tested with go example, seems like providing command through LoadTest configuration file override the ENTRYPOINT. (note that "The command field corresponds to entrypoint.")
"The command and arguments that you define in the configuration file override the default command and arguments provided by the container image. If you define args, but do not define a command, the default command is used with your new arguments."  https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/

Reply:
Ok, thanks for investigating this. What we have now seems good enough, so I'm going to approve.
We should keep our eyes open for possible simplifications in the future though. The current version is not as easy to maintain as I'd wish.